### DISCUSSION TITLE
Event location photos.

### DISCUSSION THREAD

**Discussion by vort3** [kuzmoyev/google-calendar-simple-api#122] [General]
Ugh, I just typed a lot but didn't know that I have to choose a category first, then after choosing a category I lost all my text… Now I'll have to retype everything again. Did not expect this from GitHub.

So, long story short this time:
When user creates an event manually (in google calendar android app), after they enter event location, they get a list of search results for that location, and when they choose one from that list, their event gets a nice location photo attached (I guess it's being pulled from google maps). The problem is, even if the location user entered is 100% the same as search result location, photo won't be attached unless user manually chose their location from search results.

Is there a way to fix this, so I get nice looking events in my calendar when I create them using gcsa?

I could in theory create some template events, manually choose their locations to get photo attached, then duplicate those events and update their details instead of creating new events, but that sounds clunky.

**Comment by kuzmoyev** [comment]
Hmm, good question. Actually, simple duplication of the event doesn't do it for me either. And creating the event with a third-party app doesn't generate the image. I've checked the contents of two events (created with the google calendar app and duplicated with gcsa), they seem to be the same (apart from creation time and id). Couldn't find any topics on this.

So at this point, unfortunately, don't have a solution. There might not be one for the API. Let's have this discussion open. If you find anything, please do share it here.

**Reply by vort3** [reply]
Well, I didn't try to duplicate existing events, so thanks for saving me some time, I guess.

I think even Google don't have anything about event photos in their API docs, might be that each event has some link to google maps location that users can't get or set via API, only google calendar app can do that. And that's not really fair, if google limit their API intentionally to make their app look better than any of the alternatives by using some internal hacks.

For now I just manually open each generated event, tap on a location and «choose» the only possible location result to get a photo…

**Reply by vort3** [reply]
Duplicating events in google calendar app also duplicates the image, so it's certainly tied to event somehow…
